lib/astarte_flow_web/controllers/flow_controller.ex

```diff
@@ -35,6 +36,11 @@ defmodule Astarte.FlowWeb.FlowController do
       |> put_status(:created)
       |> put_resp_header("location", Routes.flow_path(conn, :show, realm, flow))
       |> render("show.json", flow: flow)
+    else
+      {:error, %Error{} = reason} ->
+        conn
+        |> put_status(:unprocessable_entity)
+        |> render("error.json", error: reason)
```

Comment:
You need to also add a clause for every other error that goes to FallbackController
```
{:error, reason} ->
  {:error, reason}
```